### PR TITLE
Upgrade Winstone 5.19 which include Jetty 9.4.42.v20210604

### PR DESCRIPTION
**Pull request by olamy**
Signed-off-by: Olivier Lamy <olamy@apache.org>

### Proposed changelog entries

* Winstone 5.19: Update Jetty from 9.4.41.v20210516 to Jetty 9.4.42.v20210604
    * Winstone 5.19 changelog: https://github.com/jenkinsci/winstone/releases/tag/winstone-5.19
    * Jetty 9.4.42 changelog: https://github.com/eclipse/jetty.project/releases/tag/jetty-9.4.42.v20210604

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

N/A

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by daniel-beck** [comment]
Do we expect problems from the `commons-io` update? Probably doesn't matter?

**Comment by olamy** [comment]
@daniel-beck I don't think so. `commons-io` usage in `winstone` is very limited (could be removed probably)

**Comment by daniel-beck** [comment]
Oops, only a test dependency anyway.

**Review by daniel-beck — APPROVED** [review]
🤷

**Review by basil — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
